lib/resty/openssl/version.lua

```diff
@@ -19,81 +19,83 @@ local types_table
 
 -- >= 1.1
 local ok, version_num = pcall(function()
-  local num = C.OpenSSL_version_num()
-  version_func = C.OpenSSL_version
-  types_table = {
-    VERSION = 0,
-    CFLAGS = 1,
-    BUILT_ON = 2,
-    PLATFORM = 3,
-    DIR = 4,
-    ENGINES_DIR = 5,
-    VERSION_STRING = 6,
-    FULL_VERSION_STRING = 7,
-    MODULES_DIR = 8,
-    CPU_INFO = 9,
-  }
-  return num
-end)
-
-
-if not ok then
-  -- 1.0.x
-  local _
-  _, version_num = pcall(function()
-    local num = C.SSLeay()
-    version_func = C.SSLeay_version
+    local num = C.OpenSSL_version_num()
+    version_func = C.OpenSSL_version
     types_table = {
-      VERSION = 0,
-      CFLAGS = 2,
-      BUILT_ON = 3,
-      PLATFORM = 4,
-      DIR = 5,
+        VERSION = 0,
+        CFLAGS = 1,
+        BUILT_ON = 2,
+        PLATFORM = 3,
+        DIR = 4,
+        ENGINES_DIR = 5,
+        VERSION_STRING = 6,
+        FULL_VERSION_STRING = 7,
+        MODULES_DIR = 8,
+        CPU_INFO = 9,
     }
     return num
-  end)
+end)
+
+if not ok then
+    -- 1.0.x
+    local _
+    ok, version_num = pcall(function()
+        local num = C.SSLeay()
+        version_func = C.SSLeay_version
+        types_table = {
+            VERSION = 0,
+            CFLAGS = 2,
+            BUILT_ON = 3,
+            PLATFORM = 4,
+            DIR = 5,
+        }
+        return num
+    end)
 end
 
-if not version_num or version_num < 0x10000000 then
-  error(string.format("OpenSSL version %x is not supported", version_num or 0))
+-- since pcall can return string in the case of error it must be handled appropriately
+if not ok then
+    error(string.format("OpenSSL has encountered an error: %s", tostring(version_num)))
```

Comment:
let's added the context here, how about `OpenSSL has encountered an error getting version number`?

Reply:
Reverted indentation changes. As for "OpenSSL has encountered an error getting version number", "ok" case should cover it, since in both cases getting version number is wrapped in pcall